From 8ccc92ba77b47e21ad0a1543228c9a20a2954e4f Mon Sep 17 00:00:00 2001
From: Caleb Xu <calebcenter@live.com>
Date: Wed, 22 Dec 2021 11:19:20 -0500
Subject: [PATCH] Include opencl_dep as a dependency in gegl_lib target

Follow-up to 392e60be7371362c1149a9825d7644df86416692.

In the above commit, the opencl_dep dependency was declared but
this dependency was never added to the gegl_lib target, which
needs it. As such, the addition was a no-op.

This commit adds the opencl_dep dependency to the gegl_lib target
and keeps only the opencl_headers sources, since the gegl_sources
are already included in the gegl_lib target.

Origin: Upstream
Upstream-Status: Accepted [https://gitlab.gnome.org/GNOME/gegl/-/issues/288]
---
 gegl/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/gegl/meson.build b/gegl/meson.build
index 86c7a6fa1..5f459491a 100644
--- a/gegl/meson.build
+++ b/gegl/meson.build
@@ -92,8 +92,7 @@ install_headers(gegl_headers,
 gegl_ldflags = os_osx ? ['-framework', 'OpenCL'] : []
 
 opencl_dep = declare_dependency(
-    sources : [opencl_headers],
-    link_with : [gegl_sources],
+  sources : [opencl_headers],
 )
 
 gegl_lib = library(api_name,
@@ -105,6 +104,7 @@ gegl_lib = library(api_name,
     gio,
     math,
     gmodule,
+    opencl_dep,
   ],
   c_args: gegl_cflags,
   link_args: gegl_ldflags,
-- 
GitLab

